[beta] Prepare Rust 1.89.0

cuviper opened the pull request:
- Update version placeholders
- Update channel to beta
- Account for beta revisions when normalizing versions rust-lang/rust#142930

r? cuviper

cuviper committed:
Update version placeholders

cuviper committed:
Update channel to beta

bors commented:
:pushpin: Commit 9909800de3eaf19882f8d30194dca5aa5dad813a has been approved by `cuviper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=cuviper 9909800de3eaf19882f8d30194dca5aa5dad813a -->
<!-- homu: {"type":"Approved","sha":"9909800de3eaf19882f8d30194dca5aa5dad813a","approver":"cuviper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15830956179/job/44623266206) [(plain)](https://github.com/rust-lang/rust/commit/9909800de3eaf19882f8d30194dca5aa5dad813a/checks/44623266206/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

4 LL | #![feature(no_stack_check)]
5    |            ^^^^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.0.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/40110> for more information
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/40110> for more information
8 
9 error: aborting due to 1 previous error
10 

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/40110> for more information
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/40110> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args deprecation/deprecated_no_stack_check.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/deprecation/deprecated_no_stack_check.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/deprecation/deprecated_no_stack_check" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/deprecation/deprecated_no_stack_check.rs:4:12
   |
LL | #![feature(no_stack_check)]
   |            ^^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/40110> for more information

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0557`.
------------------------------------------
---

4 LL | #![feature(no_coverage)]
5    |            ^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.74.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/114656> for more information
+    = note: removed in 1.74.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/114656> for more information
8    = note: renamed to `coverage_attribute`
9 
10 error[E0658]: the `#[coverage]` attribute is an experimental feature

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.74.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/114656> for more information
+    = note: removed in 1.74.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/114656> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args feature-gates/feature-gate-coverage-attribute.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/feature-gates/feature-gate-coverage-attribute.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/feature-gate-coverage-attribute" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/feature-gates/feature-gate-coverage-attribute.rs:4:12
   |
LL | #![feature(no_coverage)] //~ ERROR feature has been removed [E0557]
   |            ^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.74.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/114656> for more information
   = note: renamed to `coverage_attribute`

error[E0658]: the `#[coverage]` attribute is an experimental feature
##[error]  --> /checkout/tests/ui/feature-gates/feature-gate-coverage-attribute.rs:12:1
   |
LL | #[coverage(off)] //~ ERROR the `#[coverage]` attribute is an experimental feature
   | ^^^^^^^^^^^^^^^^
   |
   = note: see issue #84605 <https://github.com/rust-lang/rust/issues/84605> for more information
   = help: add `#![feature(coverage_attribute)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0557, E0658.
For more information about an error, try `rustc --explain E0557`.
---

16 LL | #![feature(test_removed_feature)]
17    |            ^^^^^^^^^^^^^^^^^^^^ feature has been removed
18    |
-    = note: removed in 1.0.0 (you are using $RUSTC_VERSION)
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23))
20 
21 error: malformed `feature` attribute input
22   --> $DIR/gated-bad-feature.rs:7:1

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23))
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23))


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args feature-gates/gated-bad-feature.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/feature-gates/gated-bad-feature.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/gated-bad-feature" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0556]: malformed `feature` attribute input
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:2:25
   |
LL | #![feature(foo_bar_baz, foo(bar), foo = "baz", foo)]
   |                         ^^^^^^^^ help: expected just one word: `foo`

error[E0556]: malformed `feature` attribute input
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:2:35
   |
LL | #![feature(foo_bar_baz, foo(bar), foo = "baz", foo)]
   |                                   ^^^^^^^^^^^ help: expected just one word: `foo`

error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:9:12
   |
LL | #![feature(test_removed_feature)] //~ ERROR: feature has been removed
   |            ^^^^^^^^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23))

error: malformed `feature` attribute input
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:7:1
   |
LL | #![feature] //~ ERROR malformed `feature` attribute
   | ^^^^^^^^^^^ help: must be of the form: `#![feature(name1, name2, ...)]`

error: malformed `feature` attribute input
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:8:1
   |
LL | #![feature = "foo"] //~ ERROR malformed `feature` attribute
   | ^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#![feature(name1, name2, ...)]`

error[E0635]: unknown feature `foo_bar_baz`
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:2:12
   |
LL | #![feature(foo_bar_baz, foo(bar), foo = "baz", foo)]
   |            ^^^^^^^^^^^

error[E0635]: unknown feature `foo`
##[error]  --> /checkout/tests/ui/feature-gates/gated-bad-feature.rs:2:48
   |
LL | #![feature(foo_bar_baz, foo(bar), foo = "baz", foo)]
   |                                                ^^^

error: aborting due to 7 previous errors

Some errors have detailed explanations: E0556, E0557, E0635.
---

4 LL | #![feature(external_doc)]
5    |            ^^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.54.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/85457> for more information
+    = note: removed in 1.54.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/85457> for more information
8    = note: use #[doc = include_str!("filename")] instead, which handles macro invocations
9 
10 error: unknown `doc` attribute `include`

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.54.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/85457> for more information
+    = note: removed in 1.54.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/85457> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args feature-gates/removed-features-note-version-and-pr-issue-141619.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/feature-gates/removed-features-note-version-and-pr-issue-141619.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/removed-features-note-version-and-pr-issue-141619" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/feature-gates/removed-features-note-version-and-pr-issue-141619.rs:3:12
   |
LL | #![feature(external_doc)] //~ ERROR feature has been removed
   |            ^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.54.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/85457> for more information
   = note: use #[doc = include_str!("filename")] instead, which handles macro invocations

error: unknown `doc` attribute `include`
##[error]  --> /checkout/tests/ui/feature-gates/removed-features-note-version-and-pr-issue-141619.rs:4:8
   |
LL | #![doc(include("README.md"))] //~ ERROR unknown `doc` attribute `include`
   |        ^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[deny(invalid_doc_attributes)]` on by default

error: aborting due to 2 previous errors
---

4 LL | #![feature(macro_reexport)]
5    |            ^^^^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.0.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/49982> for more information
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/49982> for more information
8    = note: subsumed by `pub use`
9 
10 error: cannot find attribute `macro_reexport` in this scope

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/49982> for more information
+    = note: removed in 1.0.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/49982> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args macros/macro-reexport-removed.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/macros/macro-reexport-removed.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/macros/macro-reexport-removed" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/macros/macro-reexport-removed/auxiliary"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/macros/macro-reexport-removed.rs:4:12
   |
LL | #![feature(macro_reexport)] //~ ERROR feature has been removed
   |            ^^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.0.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/49982> for more information
   = note: subsumed by `pub use`

error: cannot find attribute `macro_reexport` in this scope
##[error]  --> /checkout/tests/ui/macros/macro-reexport-removed.rs:6:3
   |
LL | #[macro_reexport(macro_one)] //~ ERROR cannot find attribute `macro_reexport` in this scope
   |   ^^^^^^^^^^^^^^ help: a built-in attribute with a similar name exists: `macro_export`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0557`.
------------------------------------------
---

4 LL | #![feature(doc_keyword)]
5    |            ^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.58.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/90420> for more information
+    = note: removed in 1.58.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
8    = note: merged into `#![feature(rustdoc_internals)]`
9 
10 error[E0557]: feature has been removed

13 LL | #![feature(doc_primitive)]
14    |            ^^^^^^^^^^^^^ feature has been removed
15    |
-    = note: removed in 1.58.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/90420> for more information
+    = note: removed in 1.58.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
17    = note: merged into `#![feature(rustdoc_internals)]`
18 
19 error: aborting due to 2 previous errors

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.58.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
-    = note: removed in 1.58.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
+    = note: removed in 1.58.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
+    = note: removed in 1.58.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args rustdoc/renamed-features-rustdoc_internals.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/rustdoc/renamed-features-rustdoc_internals.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/rustdoc/renamed-features-rustdoc_internals" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/rustdoc/renamed-features-rustdoc_internals.rs:3:12
   |
LL | #![feature(doc_keyword)] //~ ERROR
   |            ^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.58.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
   = note: merged into `#![feature(rustdoc_internals)]`

error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/rustdoc/renamed-features-rustdoc_internals.rs:4:12
   |
LL | #![feature(doc_primitive)] //~ ERROR
   |            ^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.58.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/90420> for more information
   = note: merged into `#![feature(rustdoc_internals)]`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0557`.
------------------------------------------
---

4 LL | #![feature(const_trait_impl, effects)]
5    |                              ^^^^^^^ feature has been removed
6    |
-    = note: removed in 1.84.0 (you are using $RUSTC_VERSION); see <https://github.com/rust-lang/rust/pull/132479> for more information
+    = note: removed in 1.84.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/132479> for more information
8    = note: removed, redundant with `#![feature(const_trait_impl)]`
9 
10 error[E0049]: associated function `compute` has 0 type parameters but its trait declaration has 1 type parameter

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.84.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/132479> for more information
+    = note: removed in 1.84.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/132479> for more information


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args traits/const-traits/const-trait-impl-parameter-mismatch.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/traits/const-traits/const-trait-impl-parameter-mismatch.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/traits/const-traits/const-trait-impl-parameter-mismatch" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "-Znext-solver"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/traits/const-traits/const-trait-impl-parameter-mismatch.rs:11:30
   |
LL | #![feature(const_trait_impl, effects)]
   |                              ^^^^^^^ feature has been removed
   |
   = note: removed in 1.84.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23)); see <https://github.com/rust-lang/rust/pull/132479> for more information
   = note: removed, redundant with `#![feature(const_trait_impl)]`

error[E0049]: associated function `compute` has 0 type parameters but its trait declaration has 1 type parameter
##[error]  --> /checkout/tests/ui/traits/const-traits/const-trait-impl-parameter-mismatch.rs:20:16
   |
LL |     fn compute<T: ~const Aux>() -> u32;
   |                - expected 1 type parameter
...
LL |     fn compute<'x>() -> u32 {
   |                ^^ found 0 type parameters

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0049, E0557.
For more information about an error, try `rustc --explain E0049`.
---

4 LL | #![feature(unsized_locals)]
5    |            ^^^^^^^^^^^^^^ feature has been removed
6    |
-    = note: removed in CURRENT_RUSTC_VERSION (you are using $RUSTC_VERSION)
+    = note: removed in 1.89.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23))
8    = note: removed due to implementation concerns; see https://github.com/rust-lang/rust/issues/111942
9 
10 error: aborting due to 1 previous error

Note: some mismatched output was normalized before being compared
-    = note: removed in 1.89.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23))
+    = note: removed in 1.89.0 (you are using $RUSTC_VERSION.1 (88b80702e 2025-06-23))


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args unsized-locals/yote.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/unsized-locals/yote.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/unsized-locals/yote" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0557]: feature has been removed
##[error]  --> /checkout/tests/ui/unsized-locals/yote.rs:3:12
   |
LL | #![feature(unsized_locals)] //~ERROR feature has been removed
   |            ^^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.89.0 (you are using 1.89.0-beta.1 (88b80702e 2025-06-23))
   = note: removed due to implementation concerns; see https://github.com/rust-lang/rust/issues/111942

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0557`.
------------------------------------------
```

</details>

cuviper commented:
Yeah, some `normalize-stderr` regexes are not including the last `.1` of version `1.89.0-beta.1`. I'll send a fix.

cuviper committed:
Account for beta revisions when normalizing versions

Several UI tests have a `normalize-stderr` for "you are using x.y.z"
rustc versions, and that regex is flexible enough for suffixes like
"-nightly" and "-dev", but not for "-beta.N". We can just add '.' to
that trailing pattern to include this.

(cherry picked from commit 8469966710a16fd11af832d592569bcdf2298081)

cuviper commented:
I'm jumping the process a little bit to include #142930 without waiting for master, especially because that fix is basically a no-op until we get to beta versions anyway.

@bors r+

bors commented:
:hourglass: Testing commit 7af995f461503e039b35e06837e628cad3b0e5f2 with merge 390a0ab5dc4a895235a551e502c3893c3337731d...
<!-- homu: {"type":"BuildStarted","head_sha":"7af995f461503e039b35e06837e628cad3b0e5f2","merge_sha":"390a0ab5dc4a895235a551e502c3893c3337731d"} -->
